homeassistant/components/light/rest.py

```diff
@@ -0,0 +1,440 @@
+        self._rgb_color_template = rgb_color_template
+        self._name_template = name_template
+        self._supported_features_template = supported_features_template
+        self._effect_list_template = effect_list_template
+        self._timeout = timeout
+
+    @property
+    def name(self):
+        """Return the name of the light."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._state
+
+    @property
+    def brightness(self):
+        """Read back the brightness of the light."""
+        if "brightness" in self._supported_features:
+            return self._brightness
+        return None
+
+    @property
+    def color_temp(self):
+        """Read back the color_temp of the light."""
+        if "color_temp" in self._supported_features:
+            return self._color_temp
+        return None
+
+    @property
+    def effect(self):
+        """Read back the effect of the light."""
+        if "effect" in self._supported_features:
+            return self._effect
+        return None
+
+    @property
+    def rgb_color(self):
+        """Read back the rgb_color of the light."""
+        if "rgb_color" in self._supported_features:
+            return self._rgb_color
+        return None
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        supported_features_bitfield = 0
+        if "brightness" in self._supported_features:
+            supported_features_bitfield = \
+                (supported_features_bitfield | SUPPORT_BRIGHTNESS)
+        if "color_temp" in self._supported_features:
+            supported_features_bitfield = \
+                (supported_features_bitfield | SUPPORT_COLOR_TEMP)
+        if "effect" in self._supported_features:
+            supported_features_bitfield = \
+                (supported_features_bitfield | SUPPORT_EFFECT)
+        if "rgb_color" in self._supported_features:
+            supported_features_bitfield = \
+                (supported_features_bitfield | SUPPORT_RGB_COLOR)
+        if "transition" in self._supported_features:
+            supported_features_bitfield = \
+                (supported_features_bitfield | SUPPORT_TRANSITION)
+
+        return supported_features_bitfield
+
+    @property
+    def effect_list(self):
+        """Return the list of supported effects."""
+        if "effect" in self._supported_features:
+            return self._effect_list
+        return None
+
+    @asyncio.coroutine
+    def async_turn_on(self, **kwargs):
+        """Turn the device on."""
+        body_on_t = self._body_on.async_render()
+        body_on_dict = json.loads(body_on_t)
+
+        if ATTR_BRIGHTNESS in kwargs:
+            body_brightness_t = self._body_brightness.async_render() % \
+                kwargs[ATTR_BRIGHTNESS]
+            brightness_dict = json.loads(body_brightness_t)
+            body_on_dict.update(brightness_dict)
```

Comment:
Oh no. This is not allowed.

We have a hard rule to not use templates to generate JSON. That's a big no no. This should all be removed.